streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -132,47 +130,74 @@ private static String getTaskProducerClientId(final String threadClientId, final
                 partitions
             );
 
-            if (threadProducer == null) {
```

Comment:
Factored out the following to be reused in the case of transitioning from standby; a standby has only the state manager, everything else must be created just like a new task